src/main/java/de/tum/in/www1/artemis/service/ParticipationService.java

```diff
@@ -1011,4 +1011,15 @@ private boolean userHasPermissions(StudentParticipation participation) {
     public List<StudentParticipation> findWithSubmissionsWithResultByStudentIdAndExercise(Long studentId, Set<Exercise> exercises) {
         return studentParticipationRepository.findByStudentIdAndExerciseWithEagerSubmissionsResult(studentId, exercises);
     }
+
+    /**
+     * Get a mapping of participation ids to the number of submission for each participation.
+     *
+     * @param exerciseId the id of the exercise for which to consider participations
+     * @return the number of submissions per participation in the given exercise
+     */
+    public Map<Long, Integer> countSubmissionsPerParticipationByExerciseId(long exerciseId) {
+        List<long[]> submissionCountTuples = studentParticipationRepository.countSubmissionsPerParticipationByExerciseId(exerciseId);
+        return submissionCountTuples.stream().collect(Collectors.toMap(t -> t[0], t -> Math.toIntExact(t[1])));
```

Comment:
please use a better variable name than `t` and add one natural language comment before such asa

convert the list of arrays (with 2 values: participatoinId, submissionCount) into a Map <participationIid -> submissionCount>

Reply:
I decided to name it `participationIdAndSubmissionCountPair` instead of `t` and added the following comment:

```
// convert List<[participationId, submissionCount]> into Map<participationId -> submissionCount>
```